rholang/tests/mercury-tut/coat_check_test.rho

```diff
@@ -0,0 +1,62 @@
+new testResult, updateTestResult in {
+  testResult!(true) |
+  contract updateTestResult(@bool, return) = {
+    for(r <- testResult) {
+      match [r, bool] {
+        case [true, true] => { testResult!(true) | return!(true) }
+        case _ => { testResult!(false) | return!(false) }
+      }
+    }
+  } |
+  contract @"CoatCheckDemo"(_) = new MakeCoatCheck in {
+    contract MakeCoatCheck(ret) = {
+      new port, table in {
+        ret!(*port) |
+        for (@method, @arg1, @arg2, ack <= port) {
```

Comment:
There's a cleaner method syntax:
for ("new", @arg1, ack <= port) {...}
| for ("get", @arg1, ack <= port) {...}

I'll leave it approved, but eventually we should get that fixed.